independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -577,6 +591,15 @@ private ResourceMethod createResourceMethod(ClassInfo currentClassInfo, ClassInf
                         streamElementType = defaultProducesForType[0];
                     }
                 } else if (MediaType.MULTIPART_FORM_DATA.equals(produces[0])) {
+                    if (RESPONSE.equals(nonAsyncReturnType.name())) {
```

Comment:
@geoand why is usage of `javax.ws.rs.core.Response` disallowed here?

@fedinskiy has test scenario which was using `javax.ws.rs.core.Response` without problem.

Description of the PR provides no context, thus asking here.

Reply:
Because we need the type, thus RestResponse should be used as per the error message

Reply:
Yeah, no one complained 🙂.

I guess it should have been part of the migration guide. But the fact that the error message is actionable pretty much makes it a straightforward change